main.go

```diff
@@ -231,6 +254,71 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options)
 		return err
 	}
 
+	_, resources, err := clientset.ServerGroupsAndResources()
+	for _, r := range resources {
+		if r.GroupVersion == "spire.spiffe.io/v1alpha1" {
+			for _, k := range r.APIResources {
+				setupLog.Info(fmt.Sprintf("checking kind %s", k.Kind))
+				if k.Kind == "ClusterSPIFFEID" {
+					setupLog.Info("Found ClusterSPIFFEID CRD")
+					customResourcesPresent.ClusterSpiffeIDPresent = true
+				}
+				if k.Kind == "ClusterFederatedTrustDomain" {
+					setupLog.Info("Found ClusterFederatedTrustDomain CRD")
+					customResourcesPresent.ClusterFederatedTrustDomainPresent = true
+				}
+				if k.Kind == "ClusterStaticEntry" {
+					setupLog.Info("Found ClusterStaticEntry CRD")
+					customResourcesPresent.ClusterStaticEntryPresent = true
+				}
+			}
+		}
+	}
+
+	if !customResourcesPresent.fullyInitialized() {
+		setupLog.Info("CRDs missing watching for future creation of spire-controller-manager CRDs")
+		dyn, err := dynamic.NewForConfig(config)
+		if err != nil {
+			return err
+		}
+		fac := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dyn, time.Minute, metav1.NamespaceAll, nil)
+		informer := fac.ForResource(schema.GroupVersionResource{
+			Group:    apiextensions.GroupName,
+			Version:  "v1",
+			Resource: "customresourcedefinitions",
+		}).Informer()
+
+		informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				typedObj := obj.(*unstructured.Unstructured)
+				bytes, _ := typedObj.MarshalJSON()
+
+				crd := v1.CustomResourceDefinition{}
+				json.Unmarshal(bytes, &crd)
+				setupLog.Info(fmt.Sprintf("CRD added %+s", crd.Spec.Names.Kind))
+				if crd.Spec.Names.Kind == "ClusterStaticEntry" {
+					setupLog.Info("ClusterStaticEntry CRD added")
+					customResourcesPresent.ClusterStaticEntryPresent = true
+				} else if crd.Spec.Names.Kind == "ClusterFederatedTrustDomain" {
+					setupLog.Info("ClusterFederatedTrustDomain CRD added")
+					customResourcesPresent.ClusterFederatedTrustDomainPresent = true
+				} else if crd.Spec.Names.Kind == "ClusterSPIFFEID" {
+					setupLog.Info("ClusterSPIFFEID CRD added")
+					customResourcesPresent.ClusterSpiffeIDPresent = true
+				}
+
+				if customResourcesPresent.fullyInitialized() {
```

Comment:
Should we be restarting whenever CRD presence changes, not just when they are all initialized?

Reply:
I was thinking about the present problem where we are waiting for one additional resource but perhaps we should change this to restart if anything changes... This would handle a removal case as well

Reply:
@azdagron thoughts? anything else we need to get this merged and a new release out?